checks.d/openstack.py

```diff
@@ -512,3 +650,57 @@ def check(self, instance):
         except IncompleteConfig:
             self.warning("Configuration Incomplete! Check your openstack.yaml file")
             return
+
+
+    #### Local Info accessors
+    def get_local_hypervisor(self):
+        # Look up hypervisors available filtered by my hostname
+        host = self.get_my_hostname()
+        hyp = self.get_all_hypervisor_ids(filter_by_host=host)
+        return hyp[0]
+
+    def get_local_project(self):
+        if self._tenant_id:
+            return self._tenant_id
+
+        if not (self._project_name and self._domain_id):
+            # sets auth state for future use
+            self._get_auth_scope()
+
+        _filter = "?name={0}&domain_id={1}".format(self._project_name, self._domain_id)
+        url = "{0}/{1}/{2}/{3}".format(self._get_keystone_server_url(), self.DEFAULT_KEYSTONE_API_VERSION, "projects", _filter)
+        headers = {'X-Auth-Token': self._auth_token}
+
+        projects = []
+        try:
+            resp = self._make_request_with_auth_fallback(url, headers)
+            project_details = resp.json()
+            assert len(project_details["projects"]) == 1, "Non-unique project credentials"
+            return project_details["projects"][0]
+        except Exception as e:
+            self.warning('Unable to get the list of all project ids: {0}'.format(str(e)))
+
+        return projects[0]
+
+    def get_my_hostname(self):
+        return socket.gethostname()
+
+    def get_aggregates_for_local_host(self):
+        aggregate_list = self._get_and_set_aggregate_list()
+        return aggregate_list[self.get_my_hostname()]
+
+    def get_servers_managed_by_hypervisor(self):
```

Comment:
Instead of calling the get_all_server_ids method and filtering by the host, I'd recommend calling the hypervisor servers endpoint. It is *significantly* faster.

`url = '{0}/os-hypervisors/{1}/servers'.format(self._nova_url, hypervisor_id)`